src/absences/absences.test.ts

```diff
@@ -37,7 +40,52 @@ describe('Absences', () => {
 
   test('readById', async () => {
     mock.onGet(`${readPath}/1`).reply(200, { Success: true, NumResults: 1, Results: [{}] });
-    resultSingle = absences.readById(1);
-    await resultSingle.then((result) => expect(result).toStrictEqual({}));
+    const result: Absence = await absences.readById(1);
+    expect(result).toStrictEqual({});
+  });
+
+  test('create', async () => {
+    const absence = { user_id: 1, from_date: '01-01-2020', to_date: '03-01-2020', type_id: 2, subtype_id: 3 };
+    mock.onPost(createPath).reply(...successfulResponse(absence));
+    const result: Absence = await absences.create(absence);
+    expect(result).toStrictEqual(absence);
+  });
+
+  test('update', async () => {
+    const absence = { id: 1, user_id: 1, from_date: '01-01-2020', to_date: '03-01-2020', type_id: 2, subtype_id: 3 };
+    const absenceUpdate: AbsenceUpdate = { id: 7, to_date: '05-01-2020' };
+    mock.onPut(updatePath).reply(...successfulResponse(absence));
+    expect(await absences.update(absenceUpdate)).toStrictEqual(absence);
+  });
+
+  test('delete', async () => {
+    const absence = { id: 2, user_id: 1, from_date: '01-01-2020', to_date: '03-01-2020', type_id: 2, subtype_id: 3 };
+    mock.onDelete(`${deletePath}/2`).reply(...successfulResponse(absence));
+    const result: Absence = await absences.delete(2);
+    expect(result).toStrictEqual(absence);
+  });
+
+  test('approve', async () => {
+    const absence = { id: 3, user_id: 1, from_date: '01-01-2020', to_date: '03-01-2020', type_id: 2, subtype_id: 3 };
+    const absenceApprove: AbsenceApprove = { id: 3 };
+    mock.onPut(approvePath).reply(...successfulResponse(absence));
+    const result: Absence = await absences.approve(absenceApprove);
+    expect(result).toStrictEqual(absence);
+  });
+
+  test('reject', async () => {
+    const absence = { id: 4, user_id: 1, from_date: '01-01-2020', to_date: '03-01-2020', type_id: 2, subtype_id: 3 };
+    const absenceReject: AbsenceReject = { id: 4, granted_comment: 'reasons' };
+    mock.onPut(rejectPath).reply(...successfulResponse(absence));
+    const result: Absence = await absences.reject(absenceReject);
+    expect(result).toStrictEqual(absence);
+  });
+
+  test('cancel', async () => {
+    const absence = { id: 5, user_id: 1, from_date: '01-01-2020', to_date: '03-01-2020', type_id: 2, subtype_id: 3 };
+    const absenceCancel: AbsenceCancel = { id: 5 };
+    mock.onPut(cancelPath).reply(...successfulResponse(absence));
```

Comment:
Check for the correct parameter is missing...

See timetackings tests
mock.onGet(readPath, { params: { id: '99', _op__id: 'eq' } }).reply(

Reply:
Ah, nice, added it. Did not realize this was possible, it only seems to be used in one timeTrackings test. Should we make plans to add this to all tests containing a request with params or body? 

Sadly this only gives 404 if the request does not match, I experimented with adding an assertion to the reply callback instead, so that we get the nice assertion error message containing the diff. In the end I removed it again as it is probably too much code for this small benefit but you can take a look here if you are interested: https://github.com/TimeTac/js-client-library/pull/30/commits/90edfb48486685ecddb26a5cee8074d7b77943ba